### PR TITLE
Implement external APIs in a separate app with OAuth2 security

### PR DESCRIPTION
**Pull request by gregcharles**
Implemented file (TRT) import as per contract and find jobs (by authenticated user) because I'm sure they will want to have that.

### CODE REVIEW

File: client/src/main/java/tds/teacherhandscoring/model/TeacherHandScoring.java [hunk 1]
```diff
@@ -20,7 +20,7 @@
 import javax.xml.bind.annotation.XmlElement;
 
 /**
- * Teacher Hand Scoring System configuration stored as an element in the Test Package xml file.
+ * Teacher Hand Scoring System tds.support.tool.configuration stored as an element in the Test Package xml file.
```
Comment (trapnine) [inline client/src/main/java/tds/teacherhandscoring/model/TeacherHandScoring.java]: search/replace error?

File: client/src/main/java/tds/teacherhandscoring/model/TeacherHandScoringConfiguration.java [hunk 2]
```diff
@@ -14,7 +14,7 @@
 import tds.testpackage.model.TestPackage;
 
 /**
- * Teacher hand scoring configuration data to be serialized as JSON and
+ * Teacher hand scoring tds.support.tool.configuration data to be serialized as JSON and
```
Comment (trapnine) [inline client/src/main/java/tds/teacherhandscoring/model/TeacherHandScoringConfiguration.java]: search/replace error?

File: client/src/main/java/tds/testpackage/legacy/model/legacy-testspecification.xsd [hunk 3]
```diff
@@ -525,7 +525,7 @@
                 <![CDATA[
           A complete collection of items, passages, and enemies on a test
           regarding testitem* (implying 0 or more items): It is, of course, impossible to administer a test with no items,
-          however, it may be useful to simulate an itempool for the purposes of adaptive algorithm design or blueprint configuration.
+          however, it may be useful to simulate an itempool for the purposes of adaptive algorithm design or blueprint tds.support.tool.configuration.
```
Comment (trapnine) [inline client/src/main/java/tds/testpackage/legacy/model/legacy-testspecification.xsd]: search/replace error?

File: client/src/main/java/tds/testpackage/legacy/model/legacy-testspecification1.xsd [hunk 4]
```diff
@@ -525,7 +525,7 @@
                 <![CDATA[
           A complete collection of items, passages, and enemies on a test
           regarding testitem* (implying 0 or more items): It is, of course, impossible to administer a test with no items,
-          however, it may be useful to simulate an itempool for the purposes of adaptive algorithm design or blueprint configuration.
+          however, it may be useful to simulate an itempool for the purposes of adaptive algorithm design or blueprint tds.support.tool.configuration.
```
Comment (trapnine) [inline client/src/main/java/tds/testpackage/legacy/model/legacy-testspecification1.xsd]: search/replace error?

File: client/src/main/resources/xsd/legacy-testspecification.xsd [hunk 5]
```diff
@@ -525,7 +525,7 @@
                 <![CDATA[
           A complete collection of items, passages, and enemies on a test
           regarding testitem* (implying 0 or more items): It is, of course, impossible to administer a test with no items,
-          however, it may be useful to simulate an itempool for the purposes of adaptive algorithm design or blueprint configuration.
+          however, it may be useful to simulate an itempool for the purposes of adaptive algorithm design or blueprint tds.support.tool.configuration.
```
Comment (trapnine) [inline client/src/main/resources/xsd/legacy-testspecification.xsd]: search/replace error?

File: client/src/main/resources/xsd/test-package-conversion-diff.xsd [hunk 6]
```diff
@@ -61,7 +61,9 @@ The root element of the test package. The <TestPackage> element contains an opti
                                                   <xs:element minOccurs="0"
                                                   name="TeacherHandScoring">
                                                   <xs:annotation>
-                                                  <xs:documentation>Teacher Hand Scoring System configuration. Reference https://github.com/SmarterApp/TDS_TeacherHandScoringSystem</xs:documentation>
+                                                  <xs:documentation>Teacher Hand Scoring System
+                                                      tds.support.tool.configuration. Reference
```
Comment (trapnine) [inline client/src/main/resources/xsd/test-package-conversion-diff.xsd]: search/replace error?

File: client/src/main/resources/xsd/v4-test-package.xsd [hunk 7]
```diff
@@ -588,7 +588,8 @@ The Dependency can be read as:
                             </xs:element>
                             <xs:element name="TeacherHandScoring" minOccurs="0">
                                 <xs:annotation>
-                                    <xs:documentation>Teacher Hand Scoring System configuration. Reference
+                                    <xs:documentation>Teacher Hand Scoring System tds.support.tool.configuration.
```
Comment (trapnine) [inline client/src/main/resources/xsd/v4-test-package.xsd]: search/replace error?

File: external-apis/src/main/resources/application.yml [hunk 8]
```diff
@@ -0,0 +1,68 @@
...
+# URLs for dependent services
+support-tool:
+  art-rest-url: http://localhost:8080/rest
+  assessment-url: http://localhost:32841
+  exam-url: http://localhost:8081
+  content-url:  http://localhost:32848
+  thss-api-url: http://localhost:28039/api
+  tis-api-url: http://localhost:53708/api
+  permissions-url: ${support-tool.permissions-url}
+
```
Comment (trapnine) [inline external-apis/src/main/resources/application.yml]: do we need all these values in the external api app?

Comment (gregcharles) [reply external-apis/src/main/resources/application.yml]: I fixed the issues of the word "configuration" being refactored in text strings when refactoring. Reconciled the two application.yml files in service and external-api to be identical. Updated version pom.xml version numbers to keep up with a change in develop.

**Pull request merged**